src/WebJobs.Script.WebHost/ContainerManagement/LegionContainerInitializationHostService.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public class LegionContainerInitializationHostService : LinuxContainerInitializationHostService
+    {
+        private const string ContextFile = "Context.txt";
+
+        private readonly IEnvironment _environment;
+        private readonly ILogger _logger;
+
+        public LegionContainerInitializationHostService(IEnvironment environment, IInstanceManager instanceManager,
+            ILogger<LinuxContainerInitializationHostService> logger, StartupContextProvider startupContextProvider)
+            : base(environment, instanceManager, logger, startupContextProvider)
+        {
+            _environment = environment;
+            _logger = logger;
+        }
+
+        public override Task<string> GetStartContextOrNullAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("[TEST] GetStartContextOrNullAsyncOnLegion");
+
+            string containerSpecializationContextMountPath = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerSpecializationContextVolumePath);
```

Comment:
>_environment.GetEnvironmentVariable [](http://example.com/codeflow?start=61&length=35)

What happens if this env variable isn't present ? Does it return empty string ?

Reply:
Yes, it should just return null which we have a check for